torchao/dtypes/nf4tensor.py

```diff
@@ -37,11 +41,244 @@ def decorator(func):
+
+    return [
+        NF4Tensor(
+            SubclassTensorArgs(
+                chunked_size,
+                outer_wrapper.stride(),
+                outer_wrapper.storage_offset(),
+                outer_wrapper.dtype,
+                outer_wrapper.device,
+                outer_wrapper.requires_grad,
+            ),
+            outer_wrapper.block_size,
+            outer_wrapper.n_blocks,
+            outer_wrapper.scaler_block_size,
+            quantized_scalers,
+            quantization_factor,
+            outer_wrapper.scaler_mean,
+            quantized_data,
+            outer_wrapper.nf4,
+        ) for quantized_scalers, quantization_factor, quantized_data in zip(
+            quantized_scalers_chunks, quantization_factor_chunks, quantized_data_chunks
+        )
+    ]
+
+@implements(
+    [
+        aten.new_zeros.default,
+    ]
+)
+def nf4_new_zeros(aten_op, args, kwargs=None):
+    outer_wrapper = args[0]
+    new_size = args[1]
+    new_size_dim = len(new_size)
+    if (not new_size_dim in [1, 2]) or outer_wrapper.numel() % math.prod(new_size) != 0:
+        raise NotImplementedError(f"aten.new_zeros(NF4Tensor) with new size {new_size}")
+    ratio = outer_wrapper.numel() // math.prod(new_size)
+
+    assert outer_wrapper.quantized_scalers.size(0) % ratio == 0, f"quantized_scalers.numel() must be divisible by {ratio}"
+    quantized_scalers_new_zeros = aten_op(outer_wrapper.quantized_scalers, [outer_wrapper.quantized_scalers.size(0) // ratio], **kwargs)
+
+    assert outer_wrapper.quantization_factor.size(0) % ratio == 0, f"quantization_factor.size(0) must be divisible by {ratio}"
+    quantization_factor_new_zeros = aten_op(outer_wrapper.quantization_factor, [outer_wrapper.quantization_factor.size(0) // ratio], **kwargs)
+
+    assert outer_wrapper.quantized_data.size(0) % ratio == 0, f"quantized_data.size(0) must be divisible by {ratio}"
+    quantized_data_new_zeros = aten_op(outer_wrapper.quantized_data, [outer_wrapper.quantized_data.size(0) // ratio], **kwargs)
+
+    if new_size_dim == 1:
+        new_size = (new_size[0], )
+    elif new_size_dim == 2:
+        new_size = (new_size[0], new_size[1])
+
+    return NF4Tensor(
+        SubclassTensorArgs(
+            new_size,
+            outer_wrapper.stride(),
+            outer_wrapper.storage_offset(),
+            outer_wrapper.dtype,
+            outer_wrapper.device,
+            outer_wrapper.requires_grad,
+        ),
+        outer_wrapper.block_size,
+        outer_wrapper.n_blocks,
+        outer_wrapper.scaler_block_size,
+        quantized_scalers_new_zeros,
+        quantization_factor_new_zeros,
+        outer_wrapper.scaler_mean,
+        quantized_data_new_zeros,
+        outer_wrapper.nf4,
+    )
+
+@implements(
+    [
+        aten.slice.Tensor,
+    ]
+)
+def nf4_slice(aten_op, args, kwargs=None):
+    if len(args) == 5:
+        raise NotImplementedError(f"aten.slice(NF4Tensor) with step={args[4]}")
```

Comment:
You could potentially do `f"{aten_op}({",".join(list(map(type, args)))} with ...` or some such to save some lines